tfx/components/transform/executor_test.py

```diff
@@ -41,10 +43,39 @@ class _TempPath(types.Artifact):
 # TODO(b/122478841): Add more detailed tests.
 class ExecutorTest(tft_unit.TransformTestCase):
 
-  def _get_source_data_dir(self):
-    return os.path.join(
+  @classmethod
+  def setUpClass(cls):
+    cls._temp_data_dir = tempfile.mkdtemp()
```

Comment:
use [this](https://github.com/tensorflow/tfx/blob/master/tfx/components/example_gen/import_example_gen/executor_test.py#L81) to get a temp output dir

Reply:
since setUpClass is a class method, we cant call self.get_temp_dir() to get a temp directory (so we use the tempfile method. Will add a tearDownClass for cleanup.